### PR TITLE
Co-locate metadata schema and the default metadata under the same package.

### PR DESCRIPTION
**Pull request by ozevren**
This makes moving both of them together to a top-level package easier.

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[X] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: galley/pkg/config/meta/schema/codegen/tools/staticinit.main.go [hunk 1]
```diff
@@ -14,15 +14,15 @@
...
 
 import (
 	"fmt"
 	"io/ioutil"
 	"os"
 
-	"istio.io/istio/galley/pkg/config/schema"
-	"istio.io/istio/galley/pkg/config/schema/codegen"
+	"istio.io/istio/galley/pkg/config/meta/schema"
+	codegen2 "istio.io/istio/galley/pkg/config/meta/schema/codegen"
```
Comment (sushicw) [inline galley/pkg/config/meta/schema/codegen/tools/staticinit.main.go]: Is there a reason we need the import renames here and elsewhere? Is this leftover intermediate state?

Comment (ozevren) [reply galley/pkg/config/meta/schema/codegen/tools/staticinit.main.go]: No, this is introduced by the auto-refactoring that the IDE did. I just fixed them.

File: galley/pkg/config/meta/metadata/gen.go [hunk 2]
```diff
@@ -18,14 +18,14 @@ package metadata
 //go:generate go-bindata --nocompress --nometadata --pkg metadata -o metadata.gen.go metadata.yaml
```
Comment (geeknoid) [inline galley/pkg/config/meta/metadata/gen.go]: Could you call this file something other than gen.go? We use the suffix .gen.go to indicate generated files, so we skip different linters and other things on these files. I'm worried that having a file called gen.go is just inviting confusion. generate.go would be fine instead.

Comment (ozevren) [reply galley/pkg/config/meta/metadata/gen.go]: done

**Pull request merged**